tests/unit/runloop-test.js

```diff
@@ -37,4 +37,28 @@ test('"run.next" works with arguments', function(assert) {
     obj.nextMe('wat');
     done();
   }, 20);
+});
+
+test('"run.bind" works with es6 class', function(assert) {
+  assert.expect(1);
+
+  class Foo {
+    constructor() {
+      this.prop = 'foo';
+    }
+
+    @bind nextMe = () => {
+      return this.prop;
```

Comment:
If you put a debugger here, the context of `this` is the getter `{enumerable: true, configurable: false, writable: true, initializer: ƒ}`